src/scope.ml

```diff
@@ -180,9 +192,9 @@ let scope : mode -> sig_state -> env -> p_term -> tbox = fun md ss env t ->
   (* Toplevel scoping function, with handling of implicit arguments. *)
   let rec scope : env -> p_term -> tbox = fun env t ->
     (* Extract the spine. *)
-    let (p_head, args) = Syntax.get_args t in
+    let (p_head, args) = get_args t in
     (* Scope the head and obtain the implicitness of arguments. *)
-    let h = scope_head env p_head in
+    let h = scope_head env p_head args in
```

Comment:
Here it is weird to pass an extra argument to `scope_head`. What I would do instead is add a line with something like the following (where `is_patt` is similar to `Basics.is_symb`):
```ocaml
if is_patt h && args <> [] then
  fatal t.pos "Pattern variables cannot be applied.";
```
Note that the position would be better because it would correspond to the whole application as well.

Reply:
Fixed in 5b0793f